Sources/Concrete/Controller/SpeechController.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  SpeechController.swift
+//  InstantSearch
+//
+//  Created by Robert Mogos on 18/12/2017.
+//
+
+import UIKit
+import Speech
+
+/// A controller object that manages the speech recognition to text
+/// `SpeechController` is using the Speech framework, so it can only be used with iOS 10+
+
+public typealias SpeechTextHandler = (String, Bool) -> Void
+public typealias SpeechErrorHandler = (Error?) -> Void
+
+@available(iOS 10.0, *)
+@objc public class SpeechController: NSObject, SFSpeechRecognizerDelegate {
+  
+  private let speechRecognizer: SFSpeechRecognizer
+  private var speechRequest: SFSpeechAudioBufferRecognitionRequest?
+  private var speechTask: SFSpeechRecognitionTask?
+  private let audioEngine = AVAudioEngine()
+  
+  public func isRecording() -> Bool {
+      return audioEngine.isRunning
+  }
+  
+  /// Init with a locale or use the device's default one
+  public init(locale: Locale?) {
```

Comment:
ubernit: I like to see the init function at the top after the variable declarations